File: scraper/src/page_objects/login_page.js
const LOGIN_URL = 'https://ontario.client.reservauto.net/bookCar'

const USERNAME_SELECTOR = 'input[name=Username]'
const PASSWORD_SELECTOR = 'input[name=Password]'
const SUBMIT_SELECTOR = 'button[data-testid=submit]'

export default class LoginPage {
  constructor(page) {
   this.page = page
  }

  async goto() {
    await this.page.goto(LOGIN_URL)
    await this.page.waitForSelector(USERNAME_SELECTOR)
  }

  async login({ username, password }) {
    await this.page.type(USERNAME_SELECTOR, username)
    await this.page.type(PASSWORD_SELECTOR, password)

    // Wait for typing to complete
    await new Promise(r => setTimeout(r, 1000))

    await this.page.click(SUBMIT_SELECTOR)
    await this.page.waitForNavigation()
  }
}

File: car4u/src/main/kotlin/fetch/car/pages/LoginPage.kt
package fetch.car.pages

import com.microsoft.playwright.Page

class LoginPage(private val page: Page) {
  companion object {
    const val URL = "https://ontario.client.reservauto.net/bookCar"
    const val USERNAME_SELECTOR = "input[name=Username]"
    const val PASSWORD_SELECTOR = "input[name=Password]"
    const val SUBMIT_SELECTOR = "button[data-testid=submit]"
  }

  init {
    page.navigate(URL)
  }

  fun login(username: String, password: String) {
    page.fill(USERNAME_SELECTOR, username)
    page.fill(PASSWORD_SELECTOR, password)
    page.click(SUBMIT_SELECTOR)
  }
}
